File: src/Website/Models/Isaac/IsaacResource.cs
using System.Collections.Generic;
using Website.Models.Isaac.Enums;

namespace Website.Models.Isaac
{
    public class IsaacResource
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Color { get; set; }
        public ExistsIn ExistsIn { get; set; }
        public int X { get; set; }
        public int Y { get; set; }
        public Mod FromMod { get; set; }
        public ResourceType Type { get; set; }
        public List<Description> Descriptions { get; set; }
        public IsaacResource ChallengeSpecific { get; set; }
        public GameMode AvailableIn { get; set; }
        public List<IsaacResourceTag> Tags { get; set; }
    }
}

File: src/Website/wwwroot/js/src/interfaces/isaac-resource.ts
import { Mod } from './mod';
import { Tag } from './tag'

export interface IsaacResource {
    id: string,
    name: string,
    resource_type: number,
    exists_in: number,
    x: number,
    y: number,
    w: number,
    h: number,
    game_mode: number,
    color: string,
    mod: Mod | undefined,
    tags: Array<Tag> | undefined
    display_order: number | undefined;
    difficulty: number | undefined;
}